### DISCUSSION TITLE
Inbound Channel

### DISCUSSION THREAD

**Discussion by pratyak1** [kubeedge/kubeedge#3824] [General]
When deploying and using KubeEdge along with the Agent, is there an inbound channel at play (cloud to edge)? We were looking at the keadm deploy documentation but cannot be completely sure.

We have a hard requirement that all communication must be outbound only (edge-->cloud) and we are evaluating if KubeEdge fits our needs. Kindly let us know.

**Comment by RyanZhaoXB** [comment]
hello. when an edge node joins the kubeedge, the edgecore will start a websocket connection to the cloludcore.  when this websocket connection is established, it will be a long connection which enables two-way communication between edge and cloud.

**Reply by pratyak1** [reply]
Thank you for responding @RyanZhaoXB. Is the WebSocket connection on a specific port or can this be established on standard ports used for HTTP 80/443?

**Reply by RyanZhaoXB** [reply]
the port for websocket connection can be configured in cloudcore.yaml [for example](https://github.com/kubeedge/kubeedge/blob/d4bd52cfcf070f35650e192533b13e9bc53bea32/manifests/charts/cloudcore/templates/configmap_cloudcore.yaml#L35). the port is set to 10000 by default. i'm not sure if it can be set to 80/443, more likely not worked.
